Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/debuglog/internal/DebugLoggerScope.kt

```diff
@@ -9,7 +9,8 @@ import javax.inject.Qualifier
 import kotlin.coroutines.CoroutineContext
 
 object DebugLoggerScope : CoroutineScope {
-    val dispatcher = Executors.newSingleThreadExecutor(
+    val dispatcher = Executors.newFixedThreadPool(
+        4,
```

Comment:
Why 4 here? Would it make sense to use Dispatchers.Default?

Reply:
We tested 1 to 16 threads, 4 yielded the most performance gain with the least amount of threads.
* 1 thread ~ 110ms per logline on avg
* 2 ~ 50ms
* 3  ~ 35ms
* 4 ~ 20ms
* 5 ~ 15-20ms
* 6 ~ 11-20ms
etc.

Using `Dispatchers.Default` would not be bad, but we decided against it to reduce impact on other app operations.

`Dispatchers.Default` provides a thread count equal to the number of cores, e.g. 4, if the logging mechanisms block for some reasons, the coroutines running on `Default` through out the app could starve as fewer or even no threads are available. I don't think this would happen, but just to be sure it can't and to reduce the impact of logging on the app and to make the behavior recorded by the logger more accurate to what is happening when it is not active, it is nicer to work on different threads here.

Reply:
Makes Sense! Thanks for the in-depth explanation 👍